packages/langium/src/lsp/goto.ts

```diff
@@ -60,6 +54,17 @@ export class DefaultGoToResolverProvider implements GoToResolver {
             link.source.range
         ));
     }
+
+    protected findTargetNode(sourceCstNode: CstNode, targetCstNodes: Array<{ source: CstNode, target: CstNode, targetDocument: LangiumDocument }>): void {
+        const targetNode = this.references.findDeclaration(sourceCstNode);
+        if (targetNode?.element) {
+            const targetDoc = getDocument(targetNode?.element);
+            if(targetNode && targetDoc) {
+                targetCstNodes.push({ source: sourceCstNode, target: targetNode, targetDocument: targetDoc });
+            }
+        }
+    }
```

Comment:
Suggestion: I think that this should return a target node, rather than pushing it to the input parameter. Would be a bit cleaner design. Also, you should probably extract this type into an interface:
```ts
export interface GoToLink { 
  source: CstNode
  target: CstNode
  targetDocument: LangiumDocument 
}
```

```suggestion
    protected findLink(source: CstNode): GoToLink | undefined {
        const target= this.references.findDeclaration(source);
        if (target?.element) {
            const targetDocument = getDocument(targetNode?.element);
            if (target && targetDocument) {
                return { source, target, targetDocument};
            }
        }
        return undefined;
    }
```

Reply:
Makes a lot of sense